pycbc/waveform/waveform.py

```diff
@@ -736,7 +736,7 @@ def get_fd_det_waveform(template=None, **kwargs):
     """
     input_params = props(template, **kwargs)
     if 'f_lower' not in input_params:
-        input_params['f_lower'] = -1
+        raise Exception("Must set f_lower in parameters")
```

Comment:
You shouldn't have to add an exception, just take out setting it to -1. There will be an error already if it is not set in that case.